### PR TITLE
add DYNCFG_RESET 

### PR DESCRIPTION
**Pull request by underhood**
##### Summary
Some plugins support runtime reload in which case they keep the connection to netdata intact but other than that they start from 0. Support this with DYNCFG_RESET command which when sent by plugin netdata will consider it as freshly started as far as DYNCFG goes.

##### Test Plan

You can test this PR by:
- ~rebasing after #16228~
- adding `-d` to `go.d.` command line in netdata.conf (dyncfg is enabled only in debug mode for now)
- sending SIGHUP to the go.d.plugin process (this triggers reload)

Check `api/v2/config` before and some time (to allow go.d to restart) after SIGHUP and check go.d plugin is back to alive.

##### Additional Information

<details> <summary>For users: How does this change affect me?</summary>
  
</details>

### CODE REVIEW

**Review by ilyam8 — APPROVED** [review]
approving based on manual tests

File: collectors/plugins.d/pluginsd_parser.c [hunk 1]
```diff
@@ -2474,6 +2474,25 @@ static inline PARSER_RC pluginsd_register_job(char **words __maybe_unused, size_
...
+static inline PARSER_RC pluginsd_dyncfg_reset(char **words __maybe_unused, size_t num_words __maybe_unused, PARSER *parser __maybe_unused) {
+    if (unlikely(num_words != (SERVING_PLUGINSD(parser) ? 1 : 2)))
+        return PLUGINSD_DISABLE_PLUGIN(parser, PLUGINSD_KEYWORD_DYNCFG_RESET, SERVING_PLUGINSD(parser) ? "expected 0 parameters" : "expected 1 parameter: plugin_name");
+
+    if (SERVING_PLUGINSD(parser)) {
+        unregister_plugin(parser->user.host->configurable_plugins, parser->user.cd->cfg_dict_item);
+        rrdpush_send_dyncfg_reset(parser->user.host, parser->user.cd->configuration->name);
+        parser->user.cd->configuration = NULL;
+    } else {
+        const DICTIONARY_ITEM *di = dictionary_get_and_acquire_item(parser->user.host->configurable_plugins, words[1]);
```
Comment (MrZammler) [inline collectors/plugins.d/pluginsd_parser.c:2486]: Hi Timo. Not sure, but do you need later to do an `dictionary_acquired_item_release` ?

Comment (MrZammler) [reply collectors/plugins.d/pluginsd_parser.c:2486]: Ah, sorry, `unregister_plugin` does it.

**Review by MrZammler — APPROVED** [review]

**Comment by underhood** [comment]
thx

**Pull request merged**